python/interpret-core/interpret/blackbox/permutationimportance.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2019 Microsoft Corporation
+# Distributed under the MIT software license
+
+import scipy as sp
+import numpy as np
+
+import sklearn.metrics
+
+from ..utils import gen_name_from_class, gen_local_selector
+
+from interpret.api.base import ExplainerMixin
+from interpret.api.templates import FeatureValueExplanation
+from interpret.utils import unify_predict_fn, unify_data
+from interpret.utils import gen_name_from_class, gen_global_selector
+
+
+def _order_imp(summary):
+    return summary.argsort()[..., ::-1]
+
+
+class ExplainParams:
+    CLASSES = "classes"
+    MODEL_TASK = "model_task"
+    NUM_FEATURES = "num_features"
+    EXPECTED_VALUES = "expected_values"
+    CLASSIFICATION = "classification"
+    GLOBAL_IMPORTANCE_VALUES = "global_importance_values"
+    GLOBAL_IMPORTANCE_RANK = "global_importance_rank"
+    FEATURES = "features"
+    MODEL_TYPE = "model_type"
+
+
+VALID_SKLEARN_METRICS = {'mean_absolute_error',
+                         'explained_variance_score',
+                         'mean_squared_error',
+                         'mean_squared_log_error',
+                         'median_absolute_error',
+                         'r2_score',
+                         'average_precision_score',
+                         'f1_score',
+                         'fbeta_score',
+                         'precision_score',
+                         'recall_score'}
+
+
+class PermutationImportance(ExplainerMixin):
+    available_explanations = ["global"]
+    explainer_type = "blackbox"
+
+    def __init__(
```

Comment:
Can we get documentation for functions?

Reply:
Currently interpret does not have much constructor docs. I decided to drop docs instead of highlighting the distinction on this class.

I do agree a doc addition PR should be made, but I think it should add docs to groups of explainers instead of one off additions.

Reply:
There also aren't any on functions though and I know they have those

Reply:
I found no docs in https://github.com/interpretml/interpret/blob/master/python/interpret-core/interpret/blackbox

Which docs are you referring to?

Reply:
Wow ok you're right - I could have sworn there used to be more in the form of this lonely one I found: https://github.com/interpretml/interpret/blob/master/python/interpret-core/interpret/glassbox/ebm/ebm.py#L118